### ISSUE TITLE
Tracebacks are excessively verbose.

### ISSUE THREAD

**Issue by j3pic** [ipython/ipython#13981]
Whenever an unhandled exception happens in IPython, it prints out a 5-line code snippet for EVERY. SINGLE. FRAME. The resulting output can get massive when big libraries are involved. I don't need to see 5 source lines from every file in every dependency involved in the error. Can we please have a configuration option to just show an unadorned Python stack trace when an error occurs?

**Comment by ccordoba12** [comment]
There's an option for that already: `c.InteractiveShell.xmode = 'Plain'`. How to configure IPython is described here:

https://ipython.readthedocs.io/en/stable/config/intro.html

**Comment by j3pic** [comment]
The documentation is bad.

`xmode` is mentioned at your link, but there's nothing that describes what `xmode` does, and I wasn't able to find anything anywhere that describes how to configure the traceback. I would have needed to already know that the name of the attribute is `xmode`, or that controls the "mode for the IPython exception handlers" to have any chance of finding it.

I went to a page describing all the attributes in InteractiveShell, but I only skimmed over it, seeing nothing to do with tracebacks.

Search engines can't find it at all unless you include "xmode".

**Comment by ccordoba12** [comment]
Sure, perhaps the documentation is not good, but the option is there.

**Comment by Knusper** [comment]
While studying the documentation today I came across the `%xmode` magic.  The documentation does not tell anything, only by experimenting one figures out what is going on.  

A nice setting for which I now settle is `%xmode minimal` (or `c.InteractiveShell.xmode = 'Minimal'` in `ipython.config`), because with `%tb` (or `%tb verbose`) I can get a traceback if required.

Interestingly, by toggling through the options I found also `%xmode Docs`, which is not documented at all.  And that one I don't understand.

I wanted to file a PR for the documentation, but given that there is something that is beyond my knowledge I need to ask for clarification first.

**Comment by Carreau** [comment]
The source is here:

https://github.com/ipython/ipython/blob/78d65f02c9bd62883f11905bb4977dceea387d38/IPython/core/magics/basic.py#L344-L354

Feel free to send a PR that update the docstring with more information

```
In [1]: xmode?
Docstring:
Switch modes for the exception handlers.

Valid modes: Plain, Context, Verbose, and Minimal.

If called without arguments, acts as a toggle.

When in verbose mode the value `--show` (and `--hide`)
will respectively show (or hide) frames with ``__tracebackhide__ =
True`` value set.
File:      ~/dev/ipython/IPython/core/magics/basic.py
```

**Comment by Knusper** [comment]
Thanks.  I check.

**Comment by jdtsmith** [comment]
I noticed `Docs` as a mode as well.  Apparently this is from `doctb.py`, and described as:

> A stripped down version of Verbose TB, simplified to not have too much information when running doctests

in which case this is possibly a bug:

```
In [1]: %xmode?
Docstring:
Switch modes for the exception handlers.

Valid modes: Plain, Context, Verbose, and Minimal.

If called without arguments, acts as a toggle.

When in verbose mode the value `--show` (and `--hide`)
will respectively show (or hide) frames with ``__tracebackhide__ =
True`` value set.
File:      ~/.local/share/uv/tools/ipython/lib/python3.13/site-packages/IPython/core/magics/basic.py

In [2]: %xmode
Exception reporting mode: Verbose

In [3]: %xmode
Exception reporting mode: Minimal

In [4]: %xmode
Exception reporting mode: Docs
```

**Comment by Carreau** [comment]
Yeah, it should probably be better documented. I also don't like `xmode` without argument switching mode. I think it's unnatural.
